shared/src/main/java/com/couchbase/lite/ResultSet.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) 2017 Couchbase, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.couchbase.lite;
+
+import com.couchbase.litecore.C4QueryEnumerator;
+import com.couchbase.litecore.LiteCoreException;
+
+public class ResultSet {
```

Comment:
The spec is using ResultSet. Java cannot also implement Iterator as it cannot implement hasNext().